frontend/src/components/common/PopupWindow.tsx

```diff
@@ -0,0 +1,174 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { ModalProps as BsModalProps } from 'react-bootstrap';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import styled from 'styled-components';
+
+export enum ModalSize {
+  XLARGE = 'modal-xl',
+  LARGE = 'modal-l',
+  MEDIUM = 'modal-m',
+  SMALL = 'modal-s',
+}
+
+export interface ModalProps {
+  /** Optional function to control behaviour of cancel button. Default is to close the modal. */
+  handleCancel?: Function;
+  /** Optional function to control behaviour of ok button. Default is to reload the app. */
+  handleOk?: Function;
+  /** Optional text to display on the cancel button. Default is Cancel. */
+  cancelButtonText?: string;
+  /** Optional variant that will override the default variant of warning. */
+  cancelButtonVariant?:
+    | 'link'
+    | 'warning'
+    | 'primary'
+    | 'secondary'
+    | 'success'
+    | 'danger'
+    | 'info'
+    | 'dark'
+    | 'light'
+    | 'outline-primary'
+    | 'outline-secondary'
+    | 'outline-success'
+    | 'outline-danger'
+    | 'outline-info'
+    | 'outline-dark'
+    | 'outline-light';
+  /** Optional test to display on the ok button. Default is Ok. */
+  okButtonText?: string;
+  /** Optional variant that will override the default variant of primary. */
+  okButtonVariant?:
+    | 'link'
+    | 'warning'
+    | 'primary'
+    | 'secondary'
+    | 'success'
+    | 'danger'
+    | 'info'
+    | 'dark'
+    | 'light'
+    | 'outline-primary'
+    | 'outline-secondary'
+    | 'outline-success'
+    | 'outline-danger'
+    | 'outline-info'
+    | 'outline-dark'
+    | 'outline-light';
+  /** Optional title to display - no default. */
+  title?: string;
+  /** Optional message to display - no default. */
+  message?: string | React.ReactNode;
+  /** allows the parent component to control the display of this modal.
+   * Default behaviour is to show this modal on creation and close it on button click. */
+  display?: boolean;
+  /** set the value of the externally tracked display prop above. */
+  setDisplay?: (display: boolean) => void;
+  /** optional override to control the x button in the top right of the modal. Default is to show. */
+  closeButton?: boolean;
+  /** provide the size of the modal, default width is 50.0rem */
+  modalSize?: ModalSize;
+  className?: string;
+}
+
+/**
+ * Generic Component used to display modal popups to the user.
+ * @param props customize the component with custom text, and an operation to take when the component is closed.
+ */
+const PopupWindow = (props: BsModalProps) => {
```

Comment:
this looks very similar to the generic modal implementation. Can we make the Modal more extensible or get similar functionality (like checking the handling on cancel/ok)? I am concerned that we might need to mirror fixes on both files since they look so similar

Reply:
thats fair, I'll see if I can make this happen via a prop instead.